src/error.js

```diff
@@ -289,6 +293,46 @@ function reportErrorToServer(message, filename, line, col, error) {
   }
 }
 
+/**
+ * Passes the given error data to the viewer if the following criteria is met:
+ * - The viewer is a trusted viewer
+ * - The viewer has the `errorReporter` capability
+ * - The AMP doc is in single doc mode
+ * - The AMP doc is opted-in for error interception (`<html>` tag has the
+ *   `allow-error-reporting-to-viewer` attribute)
+ *
+ * @param {!JsonObject} data Data from `getErrorReportData`.
+ * @return {!Promise<boolean|undefined>} True if the error was sent to the
+ *     viewer, `Promise<undefined>` otherwise.
+ * @visibleForTesting
+ */
+export function maybeReportErrorToViewer(data) {
+  const ampdocService = Services.ampdocServiceFor(self);
+  if (!ampdocService.isSingleDoc()) {
+    return Promise.resolve();
+  }
+  const ampdocSingle = ampdocService.getAmpDoc();
+  const htmlElement = ampdocSingle.getRootNode().documentElement;
+  const docOptedIn =
+      htmlElement.hasAttribute('allow-error-reporting-to-viewer');
+  if (!docOptedIn) {
+    return Promise.resolve();
+  }
+
+  const viewer = Services.viewerForDoc(ampdocSingle);
+  if (!viewer.hasCapability('errorReporter')) {
```

Comment:
Tangent: I don't see this updated in the internal viewer docs yet.

Reply:
Right this is ongoing on our end. Dropped you a note about this.